### ISSUE TITLE
AppImage is missing something

### ISSUE THREAD

**Issue by vanyasem** [Pext/Pext#99]
### Is this a bug or a feature request?
Bug

### What Pext version are you running? (the output of ``pext --version``)
Latest Pext (0.13) AppImage

### What OS are you using? (e.g.: Ubuntu 16.04, Windows 10, macOS High Sierra)
Arch GNU/Linux

##### What did you do to run into this bug, step-by-step?
1) Install KDE1 with Qt1 on Arch GNU/Linux
2) Download the AppImage
2) Attempt to launch it

##### What did you expect would happen?
Pext launches

##### What actually happened?
Pext refused to launch until I manually installed Qt5

I assume something is missing from the AppImage that's preventing Pext from launching on a clean system
![photo_2018-04-08_20-59-25](https://user-images.githubusercontent.com/13350458/38470868-bea1b5b2-3b58-11e8-9366-d19c18fbae26.jpg)

**Comment by vanyasem** [comment]
It appears that the AppImage is missing `libxi` [1.7.9-1](https://www.archlinux.org/packages/extra/x86_64/libxi/)

Log: 
![2018-04-11-114553_968x1308_scrot](https://user-images.githubusercontent.com/13350458/38606220-f4f95e04-3d64-11e8-875f-c0800d086770.png)

**Comment by TheLastProject** [comment]
Libxi does seem to be part of standard Ubuntu. When attempting to remove it there, it'll want to remove pretty much all of GNOME. Do you know if this also occurs on a clean Arch install? I'm wondering if the KDE1 switch would've removed it

**Comment by vanyasem** [comment]
It has nothing to do with KDE1. Install any window manager (not DE) like X11 and the error is reproducible

**Comment by TheLastProject** [comment]
I'm honestly still confused about how AppImages work exactly, but could you try this build? https://transfer.sh/6753P/Pext-x86_64.AppImage

**Comment by vanyasem** [comment]
Still broken

**Comment by TheLastProject** [comment]
@TheAssassin Do you have any clue how to deal with this?

**Comment by TheAssassin** [comment]
First of all, Arch Linux is annoying for AppImage packagers, as Arch continuously breaks compatibility. Other rolling release distros don't do that. We regularly need to be inventive and mitigate their annoying issues.

That said, let's have a look at this issue.

The first thing to check is whether libxi is meant to be excluded by checking the [excludelist](https://github.com/AppImage/AppImages/blob/master/excludelist). libxi is not included, so it should probably be bundled.

Looking at how the AppImage is built, we can see that it's basically a miniconda environment of which an AppImage has been built. We tried to include additional libraries from the system based on the excludelist, but that didn't work. Therefore, we just used whatever conda provided, and didn't fiddle with that.

Did you check whether there's known compatibility issues with miniconda or some of the third-party packages we use in combination with Arch Linux?

Looking at the [build script](https://github.com/Pext/Pext/blob/master/travis/build-appimage.sh#L37), we install the libraries from pip.

@vanyasem please clone this repository, and then run `bash travis/build-appimage.sh`. Please upload the resulting AppImage.

**Comment by vanyasem** [comment]
I got an error while building: `/usr/bin/appstreamcli: error while loading shared libraries: libselinux.so.1: cannot open shared object file: No such file or directory`.

I installed `libselinux` from AUR and the error now is: `/usr/bin/appstreamcli: symbol lookup error: /usr/lib/libappstream.so.4: undefined symbol: g_strv_contains`

What are the proper dependencies to build an AppImage?

**Comment by TheAssassin** [comment]
@vanyasem appstreamcli is an optional dependency. It's strange that it doesn't work, but it's got nothing to do with AppImage per se. If you can remove it, you should probably try to remove it and re-run the build script.

Arch Linux...

**Comment by TheAssassin** [comment]
@vanyasem I've had an appstream issue while trying to debug another issue:

```
(appstreamcli:17913): GLib-ERROR **: gmem.c:103: failed to allocate 18446744073686968913 bytes
```

[Supporting "AppStream" is an almost impossible task.](https://github.com/AppImage/AppImageKit/issues/603)

**Comment by vanyasem** [comment]
Hacked my system to create the AppImage. You can get it from here https://halium.mynameisivan.ru/Pext/

**Comment by TheAssassin** [comment]
I've been busy the last few days. I hope I get to checking this tonight.

**Comment by TheLastProject** [comment]
Fixed in 7e1e7244da0691d55216e9d3aa5854eb08092451.

**Comment by TheAssassin** [comment]
I had this on my whiteboard with low prio. I somehow remembered this to be an AppStream issue, but actually that library was missing...

Anyway, glad you found the issue, @TheLastProject. It's good to see that there's a package in the conda forge.
